File: Python/RotateK.py
class Node:
    def __init__(self,val):
        self.val = val
        self.next = None
        self.head = None
    
    def traverse(self,head):
        if head:
            print(head.val)
            self.traverse(head.next)
    
    def rotateK(self,key):
    tmp = self.head
    count = 0
    while tmp.next:
        if not tmp.next:
            tail = tmp
        count += 1
        tmp = tmp.next
    tail2id = count - key
    head2id = tail2id + 1
    head2= node(head2.val)
    tmp = head
    while tail2id != 1:
        tmp = tmp.next
        tail2id -= 1
        if tail2id == 2:
            tail2 = tmp.next
    head2 = tail2.next
    while head2:
        list2.append(head2.val)
    while head != tail2id:
        list2.append(head.val)
        head = head.next
    return list2

File: Java/RotateK.java
class RotateK {
    public static void main(String[] args) {
        int n = -4;
        System.out.println(n % 5);
        for (int i : Integer.toString(args)) {
            System.out.println(i);
        }
    }

    static class ListNode {
        int val;
        ListNode next;
        ListNode(int x) {
            val = x;
            next = null;
        }
    }
    public ListNode rotateRight(ListNode head,int n){
        if (head == null || head.next == null) return null;
        ListNode dummy = new ListNode(0);
        ListNode fast = dummy, slow = dummy;
        int length;
        for (length = 0; fast != null; length++) fast = fast.next;
        //get len - n%len th node
        for (int j = length - n % length; j > 0; j--) slow = slow.next;
        fast.next = dummy.next;
        slow.next = null;
        return dummy.next;
    }

}